File: contents/contents/forward_euler_method/code/javascript/euler.js
function forwardEuler(timeStep, n) {
  const arr = [1];
  for (let i = 1; i <= n; i++) {
    arr[i] = arr[i - 1] - 3 * arr[i - 1] * timeStep;
  }
  return arr;
}

function checkEuler(arr, timeStep, threshold) {
  let isApprox = true;
  arr.forEach((_value, i) => {
    const solution = Math.exp(-3 * timeStep * i);

    if (Math.abs(arr[i] - solution) > threshold) {
      console.log(arr[i], solution);
      isApprox = false;
    }
  });
  return isApprox;
}

function main() {
  const timeStep = 0.01;
  const threshold = 0.01;
  const n = 100;
  const eulerResult = forwardEuler(timeStep, n);
  const checkResult = checkEuler(eulerResult, timeStep, threshold);
  console.log(checkResult);
}

main();

File: contents/contents/forward_euler_method/code/julia/euler.jl
function solve_euler(timestep::Float64, n::Int64)
    euler_result = Vector{Float64}(undef, n)

    # Setting the initial condition
    euler_result[1] = 1;
    for i = 2:length(euler_result)
        euler_result[i] = euler_result[i-1] - 3.0*euler_result[i-1]*timestep
    end
    return euler_result
end

function check_result(euler_result::Vector{Float64}, threshold::Float64,
                      timestep::Float64)
    is_approx = true

    for i = 1:length(euler_result)
        time = (i - 1)*timestep
        solution = exp(-3*time);
        if (abs(euler_result[i] - solution) > threshold)
            println(euler_result[i], solution)
            is_approx = false
        end
    end

    return is_approx
end

function main()
    timestep = 0.01
    n = 100
    threshold = 0.01

    euler_result = solve_euler(timestep,n)
    is_approx = check_result(euler_result, threshold, timestep)

    println(is_approx)
end

main()
